src/dotnet-aspnet-codegenerator/CodeGenCommandExecutor.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Workspaces;
+using Microsoft.Extensions.CodeGeneration.DotNet;
+using Microsoft.Extensions.CodeGeneration.EntityFrameworkCore;
+using Microsoft.Extensions.CodeGeneration.Templating;
+using Microsoft.Extensions.CodeGeneration.Templating.Compilation;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using NuGet.Frameworks;
+
+namespace Microsoft.Extensions.CodeGeneration
+{
+    public class CodeGenCommandExecutor 
+    {
+        private ProjectContext _projectContext;
+        private string[] _codeGenArguments;
+        private string _configuration;
+        private string _nugetPackageDir;
+        private ILogger _logger;
+        
+        public CodeGenCommandExecutor(ProjectContext projectContext, string[] codeGenArguments, string configuration, string nugetPackageDir, ILogger logger)
+        {
+            if(projectContext == null) 
+            {
+                throw new ArgumentNullException(nameof(projectContext));
+            }
+            if(codeGenArguments == null)
+            {
+                throw new ArgumentNullException(nameof(codeGenArguments));
+            }
+            if(logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+            _projectContext = projectContext;
+            _codeGenArguments = codeGenArguments;
+            _configuration = configuration;
+            _logger = logger;
+            _nugetPackageDir = nugetPackageDir;
+        }
+        
+        public int Execute()
+        {
+            var serviceProvider = new ServiceProvider();
+            Directory.SetCurrentDirectory(_projectContext.ProjectDirectory);
```

Comment:
This is a remnant of some old issue, that needed the directory to be set (I am not sure, but I think it had to do something with the project exports.) should not be needed anymore. 
Will remove.